Quadriga/src/main/java/edu/asu/spring/quadriga/web/search/TextConceptSearchController.java

```diff
@@ -53,85 +52,83 @@
-                        } else {
-                            if (txtFile.getAccessibility() == ETextAccessibility.PUBLIC) {
-                                texts.add(txtFile);
-                                textFileManager.loadFile(txtFile);
-                                txtFile.setSnippetLength(40);
-                            }
-                        }
-                    }
-                }
-
-                model.addAttribute("references", handles);
-                model.addAttribute("texts", texts);
-
-                List<IProject> projects = projectManager.getProjectListByAccessibility(EProjectAccessibility.PUBLIC);
-
-                List<String> projectIds = new ArrayList<String>();
-                projects.forEach(p -> projectIds.add(p.getProjectId()));
-
-                ITransformedNetwork transformedNetwork = transformationManager
-                        .getSearchTransformedNetworkMultipleProjects(projectIds, conceptUri, INetworkStatus.APPROVED);
+        Set<String> references = new HashSet<String>();
+        if (results != null && !results.isEmpty()) {
+            ElementEventsType events = marshallingService.unMarshalXmlToElementEventsType(results);
+            List<CreationEvent> eventList = events.getRelationEventOrAppellationEvent();
+            for (CreationEvent event : eventList) {
+                String sourceRef = event.getSourceReference();
+                references.add(sourceRef);
+            }
+        }
 
-                String json = null;
-                if (transformedNetwork != null) {
-                    json = jsonCreator.getJson(transformedNetwork.getNodes(), transformedNetwork.getLinks());
+        for (String sourceRef : references) {
+            ITextFile txtFile = textFileManager.getTextFileByUri(sourceRef);
+            if (txtFile == null || txtFile.getAccessibility() == ETextAccessibility.PRIVATE) {
+                handles.add(sourceRef);
+            } else {
+                if (txtFile.getAccessibility() == ETextAccessibility.PUBLIC) {
+                    texts.add(txtFile);
+                    textFileManager.loadFile(txtFile);
+                    txtFile.setSnippetLength(40);
                 }
+            }
+        }
 
-                if (transformedNetwork == null || transformedNetwork.getNodes().size() == 0) {
-                    model.addAttribute("isNetworkEmpty", true);
-                }
+        List<String> projectIds = new ArrayList<String>();
+        ITransformedNetwork transformedNetwork = null;
+        if (conceptUriSearchList.size() >= 1) {
+            List<IProject> projects = projectManager.getProjectListByAccessibility(EProjectAccessibility.PUBLIC);
+            projects.forEach(p -> projectIds.add(p.getProjectId()));
+            transformedNetwork = transformationManager.getSearchTransformedNetworkMultipleProjects(projectIds,
+                    conceptUriSearchList, INetworkStatus.APPROVED);
+        }
```

Comment:
that doesn't make much sense; don't you call Neo4j twice? first to find the statements and then again to find all networks? since already request all statements that match your search earlier, just make sure you get the complete statements and then check if they are in a public project and approve network instead of filtering all networks again.